### PR TITLE
test-wifi: check for running wpa_supplicant, daemonize it, use hardcoded paths

### PR DESCRIPTION
**Pull request by autrimpo**

### CODE REVIEW

File: user/uipcps/test-wifi.h [hunk 1]
```diff
@@ -39,6 +39,10 @@
 
 #define RL_WIFI_SSID_LEN 129
 
+#define RL_WIFI_CONF_PATH "/etc/wpa_supplicant/rlite.conf"
```
Comment (vmaffione) [inline user/uipcps/test-wifi.h]: `RL_WPA_SUPPLICANT_CONF_PATH`, as we will probably have one for `RL_HOSTAPD_..` (just to run the daemon).

File: user/uipcps/test-wifi.h [hunk 2]
```diff
@@ -39,6 +39,10 @@
...
+#define RL_WIFI_PID_PATH "/run/wpa_supplicant.pid"
```
Comment (vmaffione) [inline user/uipcps/test-wifi.h]: `RL_WPA_SUPPLICANT_PIDFILE_PATH`

File: user/uipcps/test-wifi.c [hunk 3]
```diff
@@ -56,21 +60,91 @@ update_config=1
...
+    const int buf_size = 128;
+    char buffer[buf_size];
+    char *ctrl_dir = NULL;
+    FILE *config_fd;
+
+    config_fd = fopen(config, "r");
+    if (!config_fd) {
+        fprintf(stderr, "Could not open config file\n"
+                        "Please make sure there is a config file located at %s"
+                        " and that it is accessible\n",
```
Comment (vmaffione) [inline user/uipcps/test-wifi.c]: I feel like "and it is accessible" is redundant/obvious. What do you mean exactly?

File: user/uipcps/test-wifi.c [hunk 4]
```diff
@@ -637,10 +707,12 @@ main(int argc, char **argv)
 
     /* Cleanup. */
     destroy_net_list(&networks);
+    if (terminate) {
+        printf("Terminating wpa_supplicant\n");
+        send_cmd(ctrl_conn, "TERMINATE");
```
Comment (vmaffione) [inline user/uipcps/test-wifi.c]: So I guess this is enough, and we don't need to send it SIGTERM?

File: user/uipcps/test-wifi.c (hunk 3 above)
Comment (autrimpo) [reply user/uipcps/test-wifi.c]: I mean the user test-wifi is being ran as needs to have read permissions on the file.

File: user/uipcps/test-wifi.c (hunk 4 above)
Comment (autrimpo) [reply user/uipcps/test-wifi.c]: No, and I think using the supported command for shutting down the daemon is a better/cleaner option.

File: user/uipcps/test-wifi.c (hunk 3 above)
Comment (vmaffione) [reply user/uipcps/test-wifi.c]: ok, sure

File: user/uipcps/test-wifi.c (hunk 4 above)
Comment (vmaffione) [reply user/uipcps/test-wifi.c]: yes it's better

**Pull request merged**